### PR TITLE
Add misc TypeConverter tests and cleanup the product code

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EditorAttribute.cs [hunk 1]
```diff
@@ -27,31 +23,26 @@ public EditorAttribute()
...
         /// <summary>
-        /// <para>Initializes a new instance of the <see cref='System.ComponentModel.EditorAttribute'/> class with the type name and base type
-        ///    name of the editor.</para>
+        /// Initializes a new instance of the <see cref='System.ComponentModel.EditorAttribute'/> class with the type name and base type
+        /// name of the editor.
         /// </summary>
         public EditorAttribute(string typeName, string baseTypeName)
         {
-            string temp = typeName.ToUpper(CultureInfo.InvariantCulture);
-            Debug.Assert(temp.IndexOf(".DLL") == -1, $"Came across: {typeName} . Please remove the .dll extension");
```
Comment (hughbe) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EditorAttribute.cs]: This is a bogus assert

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EditorAttribute.cs [hunk 2]
```diff
@@ -27,31 +23,26 @@ public EditorAttribute()
...
         }
 
         /// <summary>
-        /// <para>Initializes a new instance of the <see cref='System.ComponentModel.EditorAttribute'/> class.</para>
+        /// Initializes a new instance of the <see cref='System.ComponentModel.EditorAttribute'/> class.
         /// </summary>
         public EditorAttribute(string typeName, Type baseType)
         {
-            string temp = typeName.ToUpper(CultureInfo.InvariantCulture);
-            Debug.Assert(temp.IndexOf(".DLL") == -1, $"Came across: {typeName} . Please remove the .dll extension");
```
Comment (hughbe) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EditorAttribute.cs]: This is a bogus assert

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ListChangedEventArgs.cs [hunk 3]
```diff
@@ -2,83 +2,45 @@
...
             PropertyDescriptor = propDesc;
             OldIndex = newIndex;
         }
 
-        /// <summary>
-        ///    <para>[To be supplied.]</para>
-        /// </summary>
         public ListChangedEventArgs(ListChangedType listChangedType, PropertyDescriptor propDesc)
         {
-            Debug.Assert(listChangedType != ListChangedType.Reset, "this constructor is used only for changes in the list MetaData");
```
Comment (hughbe) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ListChangedEventArgs.cs]: These are bogus asserts

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ListChangedEventArgs.cs [hunk 4]
```diff
@@ -2,83 +2,45 @@
...
             ListChangedType = listChangedType;
             PropertyDescriptor = propDesc;
         }
 
-        /// <summary>
-        ///    <para>[To be supplied.]</para>
-        /// </summary>
         public ListChangedEventArgs(ListChangedType listChangedType, int newIndex, int oldIndex)
         {
-            Debug.Assert(listChangedType != ListChangedType.PropertyDescriptorAdded, "this constructor is used only for item changed in the list");
```
Comment (hughbe) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ListChangedEventArgs.cs]: These are bogus asserts

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerVerbCollection.cs [hunk 5]
```diff
@@ -2,150 +2,61 @@
...
-        /// </summary>
-        protected override void OnRemove(int index, object value)
-        {
-        }
-        /// <summary>
-        ///    <para>[To be supplied.]</para>
-        /// </summary>
-        protected override void OnValidate(object value)
-        {
-            Debug.Assert(value != null, "Don't add null verbs!");
```
Comment (hughbe) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerVerbCollection.cs]: This is a bogus assert

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs [hunk 6]
```diff
@@ -46,38 +35,28 @@ public DesignerAttribute(Type designerType)
...
-        ///       Initializes a new instance of the <see cref='System.ComponentModel.DesignerAttribute'/> class using the designer type and the
-        ///       base class for the designer.
-        ///    </para>
+        /// Initializes a new instance of the <see cref='System.ComponentModel.DesignerAttribute'/> class using the designer type and the
+        /// base class for the designer.
         /// </summary>
         public DesignerAttribute(string designerTypeName, string designerBaseTypeName)
         {
-            string temp = designerTypeName.ToUpper(CultureInfo.InvariantCulture);
-            Debug.Assert(temp.IndexOf(".DLL") == -1, $"Came across: {designerTypeName}. Please remove the .dll extension");
```
Comment (hughbe) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs]: This is a bogus assert

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs [hunk 7]
```diff
@@ -46,38 +35,28 @@ public DesignerAttribute(Type designerType)
...
-        ///       Initializes a new instance of the <see cref='System.ComponentModel.DesignerAttribute'/> class, using the name of the designer
-        ///       class and the base class for the designer.
-        ///    </para>
+        /// Initializes a new instance of the <see cref='System.ComponentModel.DesignerAttribute'/> class, using the name of the designer
+        /// class and the base class for the designer.
         /// </summary>
         public DesignerAttribute(string designerTypeName, Type designerBaseType)
         {
-            string temp = designerTypeName.ToUpper(CultureInfo.InvariantCulture);
-            Debug.Assert(temp.IndexOf(".DLL") == -1, $"Came across: {designerTypeName}. Please remove the .dll extension");
```
Comment (hughbe) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs]: This is a bogus assert

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/InstanceDescriptor.cs [hunk 8]
```diff
@@ -150,10 +142,6 @@ public object Invoke()
             {
                 return ((FieldInfo)MemberInfo).GetValue(null);
             }
-            else
-            {
-                Debug.Fail($"Unrecognized reflection type in instance descriptor: {MemberInfo.GetType().Name}");
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/InstanceDescriptor.cs]: You removed this presumably because you were able to trigger it with a test?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs [hunk 9]
```diff
@@ -117,36 +141,30 @@ protected virtual void SetRelationship(MemberRelationship source, MemberRelation
                 throw new ArgumentException(SR.Format(SR.MemberRelationshipService_RelationshipNotSupported, sourceName, source.Member.Name, relName, relationship.Member.Name));
             }
 
-            if (_relationships == null)
-            {
-                _relationships = new Dictionary<RelationshipEntry, RelationshipEntry>();
-            }
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs]: Why did you delete this?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs [hunk 10]
```diff
@@ -160,87 +178,89 @@ public override bool Equals(object o)
...
     }
 
     /// <summary>
-    ///    This class represents a single relationship between an object and a member.
+    /// This class represents a single relationship between an object and a member.
     /// </summary>
     public readonly struct MemberRelationship
     {
+        private readonly object _owner;
+        private readonly MemberDescriptor _member;
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs]: Why did you explicitly declare these rather than letting the same fields be implicitly declared as part of the get-only auto props?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs [hunk 11]
```diff
@@ -4,409 +4,329 @@
...
-        ///     "standard" to VS and Office, they are to the Microsoft .NET Framework.
+        /// This guid corresponds to the Microsoft .NET Framework command set. This is used for Verbs. While these are not
+        /// "standard" to VS and Office, they are to the Microsoft .NET Framework.
         /// </summary>
         private static readonly Guid s_ndpCommandSet = new Guid("{74D21313-2AEE-11d1-8BFB-00A0C90F26F7}");
 
-        private const int cmdidDesignerVerbFirst = 0x2000;
-        private const int cmdidDesignerVerbLast = 0x2100;
+        private const int CmdidDesignerVerbFirst = 0x2000;
+
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs]: Why did you add a blank line?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs [hunk 12]
```diff
@@ -4,409 +4,329 @@
...
         /// </summary>
         public static readonly CommandID Undo = new CommandID(s_standardCommandSet, VSStandardCommands.cmdidUndo);
+
         /// <summary>
-        ///    <para>
-        ///       Gets the GUID/integer value pair for the MultiLevelUndo command. Read
-        ///       only.
-        ///    </para>
-        /// </summary>
+        /// Gets the GUID/integer value pair for the MultiLevelUndo command. Read only.
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs]: You deleted the closing `</summary>`.  Since you're changing the XML comments, can you validate that everything compiles successfully when you turn on the validation in the compiler?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs [hunk 13]
```diff
@@ -4,409 +4,329 @@
...
-        ///       Read only.</para>
+        /// Gets the first of a set of verbs. Read only.
         /// </summary>
-        public static readonly CommandID VerbFirst = new CommandID(s_ndpCommandSet, cmdidDesignerVerbFirst);
+        public static readonly CommandID VerbFirst = new CommandID(s_ndpCommandSet, CmdidDesignerVerbFirst);
 
         /// <summary>
-        ///    <para> Gets the last of a set of verbs.
-        ///       Read only.</para>
+        ///  Gets the last of a set of verbs.
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs]: Nit: extra space at the beginning of the text

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs [hunk 14]
```diff
@@ -2,42 +2,31 @@
...
-        ///    </para>
+        /// Initializes a new instance of the <see cref='System.ComponentModel.DesignerAttribute'/> class using the name of the type that
+        /// provides design-time services.
         /// </summary>
         public DesignerAttribute(string designerTypeName)
         {
-            string temp = designerTypeName.ToUpper(CultureInfo.InvariantCulture);
-            Debug.Assert(temp.IndexOf(".DLL") == -1, $"Came across: {designerTypeName}. Please remove the .dll extension");
-            DesignerTypeName = designerTypeName;
+            DesignerTypeName = designerTypeName ?? throw new NullReferenceException();
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs]: Ugh.  I understand why you're doing this, to preserve the previous behavior, but explicitly throwing NullReferenceException like this always looks wrong to me.  What breaks if we don't validate it and simply leave off the `?? throw new NullReferenceException`? @danmosemsft, do you have an opinion here?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs [hunk 15]
```diff
@@ -46,38 +35,28 @@ public DesignerAttribute(Type designerType)
...
-        ///    </para>
+        /// Initializes a new instance of the <see cref='System.ComponentModel.DesignerAttribute'/> class using the designer type and the
+        /// base class for the designer.
         /// </summary>
         public DesignerAttribute(string designerTypeName, string designerBaseTypeName)
         {
-            string temp = designerTypeName.ToUpper(CultureInfo.InvariantCulture);
-            Debug.Assert(temp.IndexOf(".DLL") == -1, $"Came across: {designerTypeName}. Please remove the .dll extension");
-            DesignerTypeName = designerTypeName;
+            DesignerTypeName = designerTypeName ?? throw new NullReferenceException();
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs]: Same comment/question on all of these.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs [hunk 16]
```diff
@@ -135,15 +106,14 @@ public override bool Equals(object obj)
                 return true;
             }
 
-            DesignerAttribute other = obj as DesignerAttribute;
+            if (!(obj is DesignerAttribute other))
+            {
+                return false;
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs]: Why break it apart like this, rather than just doing what the original was doing:
```C#
return obj is DesignerAttribute other && other.DesignerBaseTypeName == DesignerBaseTypeName && other.DesignerTypeName == DesignerTypeName;
```
?  FWIW, I find the extra `if` version harder to read.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EditorAttribute.cs [hunk 17]
```diff
@@ -104,15 +92,14 @@ public override bool Equals(object obj)
...
             }
 
-            EditorAttribute other = obj as EditorAttribute;
+            if (!(obj is EditorAttribute other))
+            {
+                return false;
+            }
 
-            return (other != null) && other.EditorTypeName == EditorTypeName && other.EditorBaseTypeName == EditorBaseTypeName;
+            return other.EditorTypeName == EditorTypeName && other.EditorBaseTypeName == EditorBaseTypeName;
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EditorAttribute.cs]: Same comment/question. (Will stop commenting on these.)

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ExtenderProvidedPropertyAttribute.cs [hunk 18]
```diff
@@ -4,47 +4,46 @@
...
-            e.ExtenderProperty = extenderProperty;
-            e.ReceiverType = receiverType;
-            e.Provider = provider;
+            ExtenderProvidedPropertyAttribute e = new ExtenderProvidedPropertyAttribute
+            {
+                ExtenderProperty = extenderProperty,
+                ReceiverType = receiverType,
+                Provider = provider
+            };
             return e;
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ExtenderProvidedPropertyAttribute.cs]: As long as you're changing this, why define the local at all, rather than just `return new ExtenderProvidedPropertyAttribute ...`?

File: src/System.ComponentModel.TypeConverter/src/System/Timers/TimersDescriptionAttribute.cs [hunk 19]
```diff
@@ -37,6 +36,11 @@ public override string Description
                 if (!_replaced)
                 {
                     _replaced = true;
+                    if (base.Description == null)
+                    {
+                        throw new ArgumentNullException("name");
+                    }
+
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/Timers/TimersDescriptionAttribute.cs]: Why was this added?

File: src/System.ComponentModel.TypeConverter/tests/AttributeCollectionTests.cs [hunk 20]
```diff
@@ -90,28 +87,88 @@ public void ContainsTest(int count)
...
+        [Fact]
+        public void FromExisting_NullExisting_ThrowsArgumentNullException()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("existing", () => AttributeCollection.FromExisting(null, new Attribute[0]));
+        }
+
+        [Fact]
+        public void FromExisting_NullAttributeInNewAttributes_ThrowsArgumentNullException()
+        {
+            Attribute[] existingAttributes = GetAttributes().Take(2).ToArray();
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/tests/AttributeCollectionTests.cs]: What's the reason for the Take(2) vs Take(5) vs ...?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs (hunk 14 above)
Comment (danmosemsft) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs]: I see no reason to not check for null - it's valuable because it makes throws on the problematic stack, not some time later when the field is used. But this isn't .NET Framework and we're fine with changing to ANE. I suggest this or equivalent
```DesignerTypeName = designerTypeName ?? throw new ArgumentNullExeception(nameof(designerTypeName));```
and the same in other cases.

Comment (danmosemsft) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs]: If you want you could  possibly go further and do null checks on designerBaseTypeName since at least some of the class assumes it's not null (eg GetHashCode).

**Comment by danmosemsft** [comment]
@stephentoub do you need me to review also? I won't review 8000 lines if you already did.

**Comment by stephentoub** [comment]
> do you need me to review also? I won't review 8000 lines if you already did.

I only spot-checked the tests; if you or someone else could review those, that'd be helpful.  I reviewed the product changes, and other than what I commented on, I think they're fine (of course more eyes are welcome).

**Comment by danmosemsft** [comment]
@maryamariyan could you please review the tests here, sometime this week?

**Comment by danmosemsft** [comment]
At least one test failure @hughbe  https://mc.dot.net/#/user/hughbe/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/bdc9a0273fa7b6de797c81fce0c9f8e204d696b4

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs (hunk 14 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs]: Fixed the NRE stuff. Maybe i'll submit another PR fixing the other NREs in GetHashCode once this is merged

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs (hunk 13 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs]: Fixed

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs (hunk 10 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs]: Fixed

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/InstanceDescriptor.cs (hunk 8 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/InstanceDescriptor.cs]: Can be triggered with EventInfo

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs (hunk 9 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs]: Because it is never reassigned after being initialized as a field. So after I made it `readonly` this field can never be null

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs (hunk 10 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs]: Because the layout of the struct changes after converting them to auto props.

It used to be `owner`, then `member`, but became `member` then `owner`.

This breaks the netfx tests because there is a test that checks:

```cs
[Fact]
public void Indexer_NullSourceOwner_ThrowsArgumentNullException()
{
    MemberDescriptor member = TypeDescriptor.GetProperties(typeof(TestClass))[0];

    var service = new TestMemberRelationshipService();
    Assert.Throws<ArgumentNullException>("sourceOwner", () => service[null, member]);
    Assert.Throws<ArgumentNullException>("sourceOwner", () => service[null, member] = new MemberRelationship());

    Assert.Throws<ArgumentNullException>("Owner", () => service[new FakeMemberRelationsip { Owner = null, Member = member }.ToActual()]);
    Assert.Throws<ArgumentNullException>("Owner", () => service[new FakeMemberRelationsip { Owner = null, Member = member }.ToActual()] = new MemberRelationship());
}

[Fact]
public void Indexer_NullSourceMember_ThrowsArgumentNullException()
{
    var service = new TestMemberRelationshipService();
    Assert.Throws<ArgumentNullException>("sourceMember", () => service[new object(), null]);
    Assert.Throws<ArgumentNullException>("sourceMember", () => service[new object(), null] = new MemberRelationship());

    Assert.Throws<ArgumentNullException>("Member", () => service[new FakeMemberRelationsip { Owner = new object(), Member = null }.ToActual()]);
    Assert.Throws<ArgumentNullException>("Member", () => service[new FakeMemberRelationsip { Owner = new object(), Member = null }.ToActual()] = new MemberRelationship());
}

private struct FakeMemberRelationsip
{
    public object Owner;
    public MemberDescriptor Member;

    public MemberRelationship ToActual()
    {
        MemberRelationshipWrapper wrapper = default;
        wrapper.Fake = this;
        return wrapper.Actual;
    }
}

[StructLayout(LayoutKind.Explicit)]
private struct MemberRelationshipWrapper
{
    [FieldOffset(0)] public MemberRelationship Actual;
    [FieldOffset(0)] public FakeMemberRelationsip Fake;
}
```

The reason I am using this explicit layout hack is because the indexer:
```cs
/// <summary>
/// Returns the current relationship associated with the source, or MemberRelationship.Empty if
/// there is no relationship. Also sets a relationship between two objects. Empty
/// can also be passed as the property value, in which case the relationship will
/// be cleared.
/// </summary>
[SuppressMessage("Microsoft.Design", "CA1043:UseIntegralOrStringArgumentForIndexers")]
public MemberRelationship this[MemberRelationship source]
{
    get
    {
        if (source.Owner == null)
        {
            throw new ArgumentNullException(nameof(MemberRelationship.Owner));
        }
        if (source.Member == null)
        {
            throw new ArgumentNullException(nameof(MemberRelationship.Member));
        }

        return GetRelationship(source);
    }
    set
    {
        if (source.Owner == null)
        {
            throw new ArgumentNullException(nameof(MemberRelationship.Owner));
        }
        if (source.Member == null)
        {
            throw new ArgumentNullException(nameof(MemberRelationship.Member));
        }

        SetRelationship(source, value);
    }
}

/// <summary>
/// Returns the current relationship associated with the source, or null if
/// there is no relationship. Also sets a relationship between two objects. Null
/// can be passed as the property value, in which case the relationship will
/// be cleared.
/// </summary>
[SuppressMessage("Microsoft.Design", "CA1023:IndexersShouldNotBeMultidimensional")]
public MemberRelationship this[object sourceOwner, MemberDescriptor sourceMember]
{
    get
    {
        if (sourceOwner == null)
        {
            throw new ArgumentNullException(nameof(sourceOwner));
        }
        if (sourceMember == null)
        {
            throw new ArgumentNullException(nameof(sourceMember));
        }

        return GetRelationship(new MemberRelationship(sourceOwner, sourceMember));
    }
    set
    {
        if (sourceOwner == null)
        {
            throw new ArgumentNullException(nameof(sourceOwner));
        }
        if (sourceMember == null)
        {
            throw new ArgumentNullException(nameof(sourceMember));
        }

        SetRelationship(new MemberRelationship(sourceOwner, sourceMember), value);
    }
}
```

but the constructor doesn't allow nulls:
```cs
/// <summary>
/// This class represents a single relationship between an object and a member.
/// </summary>
public readonly struct MemberRelationship
{
    private readonly object _owner;
    private readonly MemberDescriptor _member;

    public static readonly MemberRelationship Empty = new MemberRelationship();

    /// <summary>
    /// Creates a new member relationship.
    /// </summary>
    public MemberRelationship(object owner, MemberDescriptor member)
    {
        _owner = owner ?? throw new ArgumentNullException(nameof(owner));
        _member = member ?? throw new ArgumentNullException(nameof(member));
    }
}
```

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs (hunk 11 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs]: Fixed

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs (hunk 12 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs]: Fixed. I'm not sure how to turn on the validation?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs (hunk 15 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs]: Fixed

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs (hunk 16 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs]: Fixed, you're right

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ExtenderProvidedPropertyAttribute.cs (hunk 18 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ExtenderProvidedPropertyAttribute.cs]: Got it

File: src/System.ComponentModel.TypeConverter/src/System/Timers/TimersDescriptionAttribute.cs (hunk 19 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/Timers/TimersDescriptionAttribute.cs]: Reverted - the exception param name is different between netfx and netcore

File: src/System.ComponentModel.TypeConverter/tests/AttributeCollectionTests.cs (hunk 20 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/tests/AttributeCollectionTests.cs]: Absolutely none - just shorter for the test data. The reason we don't just call `ToArray` is because the data is infinite - in a while true loop yielding data

File: src/System.ComponentModel.TypeConverter/tests/CollectionChangedEventArgsTests.cs [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public class CollectionChangedEventArgsTests
+    {
+        [Theory]
+        [InlineData(CollectionChangeAction.Add, "element")]
+        [InlineData(CollectionChangeAction.Refresh + 1, null)]
+        public void Ctor_Action_Element(CollectionChangeAction action, object element)
+        {
+            var args = new CollectionChangeEventArgs(action, element);
+            Assert.Equal(action, action);
```
[the remark this answers is not in the document]
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/tests/CollectionChangedEventArgsTests.cs]: fixed, whoops!

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs (hunk 9 above)
Comment (stephentoub) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs]: Then why is the check for null earlier at https://github.com/dotnet/corefx/pull/30159/files/bdc9a0273fa7b6de797c81fce0c9f8e204d696b4#diff-8ef5408d694347692e1f4b2865cbcdbbR114 still needed?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs (hunk 12 above)
Comment (stephentoub) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs]: By turning on XML doc file generation, e.g.
https://github.com/dotnet/corefx/blob/ba496ae6a9ae055708b630b3f73395a76c2ce6ce/src/System.Threading.Tasks.Extensions/src/System.Threading.Tasks.Extensions.csproj#L6

Comment (danmosemsft) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs]: I hadn't realized that we don't have this enabled for all our assemblies. We probably should enable it globally - grandfathering in any that have invalid XML currently.

Comment (danmosemsft) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs]: ie stick it in the root dir.props

Comment (stephentoub) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs]: We would need to grandfather-in pretty much every assembly, as it'll warn-as-error for public types/members missing XML comments.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs (hunk 10 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs]: Okay. I've undone this change - also lets us remove dead code as a result

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs (hunk 9 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs]: It isn't! I've removed it

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs (hunk 12 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/StandardCommands.cs]: I turned this on for this library in this PR (I had to silence warnings for no documentation for public types etc.)

This PR is already pretty massive so I'll do that in another PR
